_posts/2020-02-09-SVGD.markdown

```diff
@@ -0,0 +1,401 @@
+          <b>3.</b> SVGD still seems to collapse in high-dimensional spaces, leading to exciting new research in <a href="https://arxiv.org/abs/1902.03394">why this occurs</a> and <a href="https://arxiv.org/abs/1711.04425">ideas on how to get around it</a>.
+     </p>
+     </details>
+  2. Prove that the discrepancy in Equation 3 of the Stein Variational Gradient Descent Paper only equals 0 when \(p\) and \(q\) are equal.
+     <details><summary>Solution</summary>
+     <p>
+      Recall the operator definition of Stein's Identity:
+
+      $$ \mathbf{E}_p[\mathcal{A}_pf(x)] = 0$$
+
+      If \( p \neq q \), we get \( \mathbf{E}_q[\mathcal{A}_pf(x)] \) for some choice of function \( f \).
+
+      We can expand this to:
+
+      $$\mathbf{E}_q[\mathcal{A}_pf(x)] - \mathbf{E}_q[\mathcal{A}_qf(x)]$$
+
+      Recalling the full definition of the operator:
+
+      $$\mathcal{A}_pf(x) = \mathbf{E}_p[s_p(x)f(x) + \nabla_x f(x)] = 0$$
+
+      where score function \( s_p(x) \) is just \( \nabla_x \log p(x) \), we are left with
+
+      $$\mathbf{E}_q[(s_p(x) - s_q(x))f(x)]$$
+
+      This means unless \(p = q \rightarrow s_p(x) = s_q(x) \; \forall x \in \mathcal{X} \), we can always find some function \(f\) for which the above quantity is nonzero.
+
+     </p>
+     </details>
+  3. Implement SVGD in your favorite language (see the notes for links to different implementations). Then, let’s take a look at the role of the kernel in SVGD:
+    
+     (a) Remove the repulsive kernel term and observe how particles collapse to modes.
+    
+     (b) Remove the kernel’s contribution in the first term.
+    
+     What happens?	
+
+<br />
+
+# 5. SVGD as Gradient Flow
+  **Motivation**: SVGD as Gradient Flow is one of the first papers that analyzes the dynamics and theoretical properties of SVGD. This paper covers an incredible amount of seemingly-disparate topics, connecting them in a succinct explanation. However, due to the relative difficulty of the material (especially regarding the background needed), the attached notes are self-contained, and can be read alongside the paper to fully understand the material.
+
+  **Topics**:
+
+  1. Weak Convergence and Convergence in Distribution
+  2. Dirac Deltas, Bounded Lipschitz Metrics, and Large Sample Regime of SVGD
+  3. Brownian Motion, Fokker-Planck, and Continuous Time Analysis of SVGD
+  4. Optimal Transport and Wasserstein Distances
+  5. Tangent Spaces, Riemmanian Metrics, and Differential Geometry Basics
+  6. SVGD as a Gradient Flow
+
+  **Notes**:  In this class, we try to understand the geometric implications of SVGD. The notes are structured relatively differently: with the amount of background needed, relevant material is introduced in-line. As a result, the ideal way to understand this week requires reading the notes alongside the paper, using the background sections to understand the concepts and their connections within the paper. See here for the notes in [PDF](/assets/svgd_notes/week05.pdf).
```

Comment:
s/:/-

Reply:
sorry, I don't really understand this notation?